Add documentation on converting from PSC to PSP

sarakonradi opened the pull request:
This PR adds a section on converting from PSC to PSP to the ``iaf_psc_exp`` models' documentation, as discussed with @jhnnsnk.

More specifically, it: 

- Integrates the [relevant documentation](https://github.com/nest/nest-simulator/blob/master/pynest/examples/Potjans_2014/helpers.py#L68) into ``iaf_psc_exp.h``
- Adds the Hanuschkin et al. (2010) reference underneath
- Directs users of other ``iaf_psc_exp`` models to this conversion documentation

Open questions: 

- Should the "Conversion from PSC to PSP" title be turned into a Level 3 heading in a separate PR? @jougs mentioned that a different PR will soon do the same with all the "Remarks". 

It fixes #1590.

clinssen commented on models/iaf_psc_exp.h:
```diff
@@ -98,6 +98,25 @@ receptor_type 1, in contrast, is filtered through an exponential
 kernel with the time constant of the excitatory synapse,
 tau_syn_ex. For an example application, see [4]_.
 
+Conversion from PSC to PSP:
```
I'm not so happy with notation here (e.g. ``v`` instead of ``V_m``, many new variables introduced but only used once). Moreover, the documentation does not actually show how to convert PSCs into PSPs. Could we replace this whole block with a reference to the postsynaptic_potential_to_current function instead?

https://github.com/nest/nest-simulator/blob/6f48c7f23f675c08cbb6d696bf382ab80d51ccda/pynest/examples/Potjans_2014/helpers.py#L65

clinssen replied:
@jhnnsnk: just wanted to check that you intended to define the step function in reverse from the common definition, i.e. Theta(t) = 1 for t <= 0?

jhnnsnk replied:
@clinssen No, that looks like a typo. Thanks for spotting it! The `\leq` should be replaced by `\geq` in https://github.com/nest/nest-simulator/blob/6f48c7f23f675c08cbb6d696bf382ab80d51ccda/pynest/examples/Potjans_2014/helpers.py#L75  @sarakonradi Could this be corrected with this PR as well?

jhnnsnk replied:
> Could we replace this whole block with a reference to the postsynaptic_potential_to_current function instead?

I really like @clinssen 's suggestion to just use a reference to the function. In particular because it avoids duplication of text, but also because knowing about the conversion is not essential for the neuron model (it is just additional information). How about just adding a sentence like: "For conversion between postsynaptic potentials (PSPs) and PSCs, please refer to the function ..." ?

sarakonradi replied:
> I really like @clinssen 's suggestion to just use a reference to the function. In particular because it avoids duplication of text, but also because knowing about the conversion is not essential for the neuron model (it is just additional information). How about just adding a sentence like: "For conversion between postsynaptic potentials (PSPs) and PSCs, please refer to the function ..." ?

Done. I agree with both of you that it would make sense to remove duplicate section in the neuron model, replacing it with a short reference to the function in ``helpers.py``.

sarakonradi commented on models/iaf_psc_exp_htum.h:
```diff
@@ -89,8 +89,11 @@ matrix objects.
 tau_syn_in, respectively, to avoid numerical instabilities.
-For details, please see IAF_neurons_singularity.ipynb in
-the NEST source code (docs/model_details).
+For details, you can check out the `IAF_neurons_singularity.ipynb
```
Done.

sarakonradi commented:
@clinssen @jhnnsnk Thanks for your input. I added the ``postsynaptic_potential_to_current function`` reference, fixed ``\geq`` and rephrased the notebook sentence. Please check!

jhnnsnk commented on models/iaf_psc_exp.h:
```diff
@@ -134,6 +139,11 @@ References
 .. [4] Schuecker J, Diesmann M, Helias M (2015). Modulated escape from a
        metastable state driven by colored noise. Physical Review E 92:052119
        DOI: https://doi.org/10.1103/PhysRevE.92.052119
+.. [5] Hanuschkin A, Kunkel S, Helias M, Morrison A and Diesmann M (2010)
```
This reference is not needed because it is not mentioned in the text any more.

sarakonradi replied:
Done!

terhorstd commented:
merging with two positive reviews
